components/core/src/clp/MemoryMappedFileView.hpp

```diff
@@ -0,0 +1,66 @@
+#ifndef CLP_MEMORYMAPPEDFILEVIEW_HPP
+#define CLP_MEMORYMAPPEDFILEVIEW_HPP
+
+#include <cstddef>
+#include <span>
+#include <string>
+#include <string_view>
+#include <utility>
+
+#include "ErrorCode.hpp"
+#include "TraceableException.hpp"
+
+namespace clp {
+/**
+ * A class for mapping a read-only file into memory. It maintains the memory buffer created by the
+ * underlying `mmap` system call and provides methods to get a view of the memory buffer.
+ */
+class MemoryMappedFileView {
```

Comment:
I think the convention is usually for a view to be a view of some other class. Since this class owns the memory-mapped resource, I feel like calling it a view is a little misleading. I guess you want to call it a view since it only supports read-only files right now. In that case, perhaps it's better to call it a `ReadOnlyMemoryMappedFile`?

Reply:
Yeah make sense. I was about to use this name but I thought you guys may feel it's too verbose